pynbody/snapshot/__init__.py

```diff
@@ -345,6 +345,13 @@ def _get_array_with_lazy_actions(self, name):
 
         if name in list(self.keys()):
             self._dependency_tracker.touching(name)
+
+            # Ensure that any underlying dependencies on 1D positions and velocities
+            # are forwarded to 3D dependencies as well
+            nd_name = self._array_name_1D_to_ND(name)
+            if nd_name is not None:
+                self._dependency_tracker.touching(nd_name)
```

Comment:
Is this right to duplicate the effort, or should this be _instead_ of `touching(name)` above?

Reply:
I am not sure. I ended up coding it as an addition for two reasons: 

(i) my pre-designed tests for the behaviour I would expect were not passing when trying dependencies only on 3D names (your _instead_ solution). I didn't fully understand why, but some other arrays would then end up not being updated following coordinate transforms.

(ii) I agree the current solution (ensuring that 3D arrays have at least all dependencies of their 1D arrays)  is safer but requires slightly more effort. However since most positions arrays end up `touching` nearly every other (which is the reason why the code is currently almost perfectly working), the current solution only adds one or two additional checks to the existing ones, so it's not "duplicating" much work.

Reply:
OK, let's accept it as is as one day (haha) the whole snapshot class could do with an overhaul